Add legend.labelcolor in rcParams (#20084)
matplotlib/matplotlib: doc/users/next_whats_new/rcparams_legend.rst

New rcParams for legend: set legend labelcolor globally
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The new :rc:`legend.labelcolor` works as a global 'labelcolor' argument in the 
> `.pyplot.legend`. Thus, the `legend.labelcolor` allows valid color string
> (e. g., 'black') toggle. Alternatively use 'linecolor','markerfacecolor'

Reviewer: ```suggestion
The new :rc:`legend.labelcolor` works as a global *labelcolor* argument in the 
`.pyplot.legend`. Thus, ``legend.labelcolor`` allows valid color string
(e.g., 'black') toggle. Alternatively use 'linecolor', 'markerfacecolor'
```

New rcParams for legend: set legend labelcolor globally
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The new :rc:`legend.labelcolor` works as a global *labelcolor* argument in the
> `.pyplot.legend`. Thus, ``legend.labelcolor`` allows valid color string
> (e.g., 'black') toggle. Alternatively use 'linecolor', 'markerfacecolor'
> (or 'mfc'), or 'markeredgecolor' (or 'mec') to match the color of the line marker

Reviewer 2: ```suggestion
A new :rc:`legend.labelcolor` sets the default *labelcolor* argument for
`.Figure.legend`.  The special values  'linecolor', 'markerfacecolor'
(or 'mfc'), or 'markeredgecolor' (or 'mec') will cause the legend text to match 
the corresponding color of marker. 
```

> New rcParams for ledend: set legend labelcolor globaly
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
New rcParams for legend: set legend labelcolor globally
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

> The new :rc:`legend.labelcolor` allows toggling valid color string(for example, 'black'),

Reviewer: ```suggestion
The new :rc:`legend.labelcolor` allows toggling valid color string (e.g., 'black'),
```

versus choosing to match the color of the line of marker using 'linecolor',

> 'markerfacecolor'(or 'mfc'), or 'markeredgecolor' (or 'mec').

Reviewer: Mention that this is related to *labelcolor* argument.

.. plot::

    import numpy as np

>     import matplotlib.pyplot as plt

Reviewer: Not needed in `.. plot::`

    
    # Make some fake data.
    a = np.arange(0, 3, .02)
    c = np.exp(a)
    d = c[::-1]
    plt.rcParams['legend.labelcolor'] = 'linecolor'

> 

Reviewer: This is the important line, so I would put it first.

Also, only one blank line.

    fig, ax = plt.subplots()
    ax.plot(a, c, 'g--', label='Model length')
    ax.plot(a, d, 'r:', label='Data length')

>     legend = ax.legend()

Reviewer: ```suggestion
    ax.legend()
```
